### DISCUSSION TITLE
Please 🙏🙏, create a version for Visual Studio 2022 Community Edition

### DISCUSSION THREAD

**Discussion by mvasquez05** [cline/cline#527] [Feature Requests]
I would love to use your fantastic "Cline" AI assistant, but my boss only allows us to use Visual Studio 2022 Community Edition. So please, if possible, try to create a version for Visual Studio 2022 Community Edition. Thank You 😀

**Comment by devMls** [comment]
+1 to this.

I test the plugin in visual studio code, with Sonnet, and with 1 euro I achieve generate template for a website (php) .

Usually i develop with c# and razor pages. 

For this test I have migrate the template to razor and I have create the backend. But with a VS plugin i will be able to integrate the process and ask for razor pages code directly

**Comment by Shane32** [comment]
Agree; really need this for VS 2022.

**Reply by christianzbox** [reply]
I also agree.  We really really need this.

**Comment by Raylopi** [comment]
That would be wonderful; many developers use more than just vscode for their projects. For those using Visual Studio 2022 with big projects in C# (or other languages) it would be a great help.

**Comment by sirinath** [comment]
Looking forward to this integration.

**Comment by tolgakisaogullari** [comment]
+1 looking forward

**Comment by CanIraktan** [comment]
+1'm waiting too

**Comment by woodysseus** [comment]
+1 looking forward to

**Comment by panupongkongkate** [comment]
+1

**Comment by auderis** [comment]
+1 looking forward

**Comment by zzmzaizai** [comment]
+1

**Reply by Lennartos** [reply]
+1

**Comment by kenifranz** [comment]
Is it out yet?

**Comment by kharonx** [comment]
+1

**Comment by davidbenichou** [comment]
+10!!!

**Comment by amishsthapit** [comment]
+100!!!

**Comment by christianzbox** [comment]
Is it out yet?

**Comment by MuhammadFaizanKhan** [comment]
Any update?

**Comment by tejpaldev** [comment]
+1 :)

**Comment by SkyGiovak** [comment]
+1

**Comment by nifh80s** [comment]
+1 :)

**Comment by yanuar-tess** [comment]
+1 ⬆️

**Comment by LZappM** [comment]
+1

**Comment by Asrafullislamriad** [comment]
+5 please please ,

**Comment by Asrafullislamriad** [comment]
+1

**Comment by Asrafullislamriad** [comment]
+5 please please ,

**Comment by wellbritto98** [comment]
+1

**Comment by PeterJia** [comment]
metoo

**Comment by supermuskk** [comment]
me too

**Comment by InJJWeTrust** [comment]
Would absolutely love to see this as coding  in Visual Studio 2022 CE is way more enticing to me than CSCode as a user.
This would definitely increase your user-base :)

**Comment by arturmoreira** [comment]
+1

**Comment by maksgranko** [comment]
+1

**Comment by linleyan** [comment]
+1

**Comment by BuyukBang** [comment]
+1

**Comment by Z-a-r-a-k-i** [comment]
+1

**Comment by matthy123** [comment]
+1 🙏

**Comment by PeterJia** [comment]
+10086

**Comment by MagusFalsus** [comment]
+1

**Comment by crmoschini** [comment]
+1

**Comment by dangiulli** [comment]
seguo

**Reply by alexander-kucherov** [reply]
+1

**Comment by rifkisururi** [comment]
+1

**Comment by MaksimAdamenko** [comment]
+3,1415

**Comment by alelom** [comment]
This would be amazing for our company too. Please consider this.

**Comment by dmitriykotik** [comment]
+1

**Comment by SLenik** [comment]
+1

**Comment by KSVTL** [comment]
+1

**Comment by Cyph3r** [comment]
+1

**Comment by Erdnussa** [comment]
+1

**Comment by 8ternity** [comment]
+1 please

**Comment by S1ckZer** [comment]
+1

**Comment by zara-tec** [comment]
+1

**Comment by ibbobud** [comment]
lets do it! im using VS Code but corp is strongly suggesting i need to use VS instead.

**Comment by tomaszgolebiowski** [comment]
+1 looking forward

**Comment by PuristCoder** [comment]
+1

**Comment by DiegoResnik** [comment]
+1

**Comment by HirunaPro** [comment]
Yes please, we are waiting

**Comment by ChrisOMetz** [comment]
Please support all editions of Visual Studio 2022.

**Comment by darksody** [comment]
+1

**Comment by Ophir333** [comment]
+= std::vector<int> please = {1, 1, 1, 1};

**Comment by jimmyhuang1980** [comment]
+1

**Comment by viacheslav-mik** [comment]
I would love to have it too.

**Comment by alekvol** [comment]
+1

**Comment by akovachev75** [comment]
+1

**Comment by Arjenlodder** [comment]
+1

**Comment by Killua-san** [comment]
+1

**Comment by IGalfi97** [comment]
+1

**Comment by softworkz** [comment]
+1

**Comment by bhmahler** [comment]
+1 Please add this :)

**Comment by luanxinchen** [comment]
+1 , plz

**Comment by madison-boman** [comment]
+1 pretty pretty pleeeeaasseee

**Comment by Davidos533** [comment]
We need this!

**Comment by KoosMos** [comment]
+1

**Comment by fahrim** [comment]
+1

**Comment by JJMLongo** [comment]
+1

**Comment by martinossx** [comment]
+1

**Comment by steven42121** [comment]
+1000!

**Comment by alex-yazdan** [comment]
+1

**Comment by Cyph3r** [comment]
I was a +1, but you really missed your opportunity here. Moved on to a better product.

**Reply by softworkz** [reply]
> Moved on to a better product.

Which one for VS?

**Reply by Cyph3r** [reply]
> > Moved on to a better product.
> 
> Which one for VS?

claude code extension itself. very good. add the memory plugin.

**Comment by dennis-8** [comment]
Please add support to Visual Studio))

**Comment by madturk2025-art** [comment]
arayüz sabit ingilizce bu kısıtlamayı kaldırırsanız mükemmel olur türkçe dil desteği ekleyebilirmisiniz arayüze isteyen istediği dili seçme özgürlüğüne sahip olsa çok iyi olurdu.

**Comment by PiotrKarczmarz** [comment]
+1 Would be cool to have Cline native support for Visual Studio 🎉

**Comment by kushalMeharwade** [comment]
Hello everyone,

Like many of you, I've been waiting for official Cline support on Visual Studio. In the meantime, I ended up building a coding agent for VS 2022/2026 with a similar workflow, in case it's useful to anyone here.

ApexCode AI Assistant — a BYOK, multi-provider AI coding assistant for Visual Studio:

Bring Your Own Key — connect OpenAI, Azure OpenAI, OpenRouter, DeepSeek, Gemini, NVIDIA, or Vercel with your own API key. No subscription, no markup.
Plan before it acts — explores your workspace read-only, proposes a plan, and waits for approval before touching any file.
Checkpoints & diffs — every AI change is snapshotted first, so you can review and revert with one click.
Roslyn-powered fixes — reads diagnostics, validates builds, and applies fixes directly to the syntax tree.
SQL Server integration — safe, read-only schema inspection and queries.

Marketplace: https://marketplace.visualstudio.com/items?itemName=KushalMeharwade.ApexCode2026&ssr=false#review-details
GitHub: https://github.com/kushalMeharwade/ApexCode
